docs/content/docs/tasks_usage/traffic_management/ingress.md

```diff
@@ -15,42 +16,187 @@ This document describes how to expose HTTP and HTTPS routes outside the cluster
 
-## Ingress controller compatibility
-Ingress in OSM is compatible with the following ingress controllers.
-- [Kubernetes Nginx Ingress Controller][2]
-- [Azure Application Gateway Ingress Controller][3]
-- [Gloo API Gateway][5]
+> Note: Enabling HTTPS ingress will disable HTTP ingress.
 
-Other ingress controllers might also work as long as they use Kubernetes Ingress resource, and allow provisioning a custom root certificate for HTTPS backend server certificate validation while using HTTPS ingress.
 
-## Ingress configurations
-The following section describes sample ingress configurations used to expose services managed by OSM outside the cluster. The configuration might differ based on the ingress controller being used.
+### Disabling HTTP or HTTPS Ingress
 
-The example configurations describe how to expose HTTP and HTTPS routes for the `bookstore-v1` service running on a pod with the service acccount `bookstore-v1` on port `14001` in the `bookstore` namespace, outside the cluster. The ingress configuration will expose the HTTP path `/books-bought` on the `bookstore-v1` service.
+#### HTTP ingress
 
-Since OSM uses its own root certificate, the ingress controller must be provisioned with OSM's root certificate to be able to authenticate the certificate presented by backend servers when using HTTPS ingress. With `Tresor` as the certificate provider, OSM stores the CA root certificate in a Kubernetes secret named `osm-ca-bundle` with the key `ca.crt` in the namespace OSM is deployed (`osm-system` by default). When using other certificate providers such as `cert-manager.io` or `Hashicorp Vault`, the `osm-ca-bundle` secret must be created by the user with the base64 encoded root certificate stored as the value to the `ca.crt` attribute in the secret's data.
+HTTP ingress can be disabled by enabling HTTPS ingress.
 
-### Using Nginx Ingress Controller
+#### HTTPs ingress
 
-#### Prerequisites
-- Install [nginx ingress controller](https://kubernetes.github.io/ingress-nginx/deploy/#installation-guide)
+Patch the ConfigMap by setting use_https_ingress: "false".
+
+> Note: Disabling HTTPS ingress will enable HTTP ingress.
+
+```bash
+# Replace osm-system with osm-controller's namespace if using a non default namespace
+kubectl patch ConfigMap osm-config -n osm-system -p '{"data":{"use_https_ingress":"false"}}' --type=merge
+```
+
+## How it works
+
+### Exposing an HTTP or HTTPS service using Ingress
+A service can expose HTTP or HTTPS routes outside the cluster using Kubernetes Ingress along with an ingress controller. Once an ingress resource is configured to expose HTTP routes outside the cluster to a service within the cluster, OSM will configure the sidecar proxy on pods to allow ingress traffic to the service based on the ingress routing rules defined by the Kubernetes Ingress resource.
+
+Note:
+1. This behavior opens up HTTP-based access to any client that is not a part of the service mesh, not just ingress.
```

Comment:
In the context of the demo, does this mean that if bookstore was exposed via ingress, then the bookthief could steal books even if SMI policies are being enforced?

Reply:
> This behavior opens up HTTP-based access to any client that is not a part of the service mesh, not just ingress.

If bookthief is not a part of the mesh, then yes it can make requests to the bookstore pod the way an ingress controller would make requests and have those requests served.

If bookthief is within the mesh, then no.